webapp/components/create_post.jsx

```diff
@@ -233,6 +233,13 @@ export default class CreatePost extends React.Component {
             return;
         }
 
+        const isDirectOrGroup = ((updateChannel.type === Constants.DM_CHANNEL) || (updateChannel.type === Constants.GM_CHANNEL));
+        if (!isDirectOrGroup && this.state.message.endsWith('/purpose ')) {
```

Comment:
Why are we using an `endsWith` here, especially with that trailing space? That should be `this.state.message.trimRight() === '/purpose'` in case people don't use the autocomplete and so that we also don't capture when people just end a sentence with `/purpose`.